includes/Framework/PaymentGateway/Payment_Gateway_Direct.php

```diff
@@ -536,91 +541,98 @@ public function get_order( $order_id ) {
+					wc_price( Money_Utility::cents_to_float( $payment->getTotalMoney()->getAmount(), $order->get_currency() ) ),
+					Payment_Gateway_Helper::payment_type_to_name( $card_type ),
+					$last_four
 				);
-			}
 
-			// adds the transaction id (if any) to the order note
-			if ( $response->get_transaction_id() ) {
-				/* translators: Placeholders: %s - transaction ID */
-				$message .= ' ' . sprintf( esc_html__( '(Transaction ID %s)', 'woocommerce-square' ), $response->get_transaction_id() );
-			}
+				// add the expiry date if it is available
+				if ( ! empty( $order->payment->exp_month ) && ! empty( $order->payment->exp_year ) ) {
 
-			/**
-			 * Direct Gateway Credit Card Transaction Approved Order Note Filter.
-			 *
-			 * Allow actors to modify the order note added when a Credit Card transaction
-			 * is approved.
-			 *
-			 * @since 3.0.0
-			 *
-			 * @param string $message order note
-			 * @param \WC_Order $order order object
-			 * @param Payment_Gateway_API_Response_Interface $response transaction response
-			 * @param Payment_Gateway_Direct $this instance
-			 */
-			$message = apply_filters( 'wc_payment_gateway_' . $this->get_id() . '_credit_card_transaction_approved_order_note', $message, $order, $response, $this );
+					$message .= ' ' . sprintf(
+						/* translators: Placeholders: %s - credit card expiry date */
+						__( '(expires %s)', 'woocommerce-square' ),
+						esc_html( $order->payment->exp_month . '/' . substr( $order->payment->exp_year, -2 ) )
+					);
+				}
 
-			$this->update_order_meta( $order, 'is_tender_type_card', true );
+				// adds the transaction id (if any) to the order note
+				if ( $response->get_transaction_id() ) {
+					/* translators: Placeholders: %s - transaction ID */
+					$message .= ' ' . sprintf( esc_html__( '(Transaction ID %s)', 'woocommerce-square' ), $response->get_transaction_id() );
+				}
 
-			$order->add_order_note( $message );
+				/**
+				 * Direct Gateway Credit Card Transaction Approved Order Note Filter.
+				 *
+				 * Allow actors to modify the order note added when a Credit Card transaction
+				 * is approved.
+				 *
+				 * @since 3.0.0
+				 *
+				 * @param string $message order note
+				 * @param \WC_Order $order order object
+				 * @param Payment_Gateway_API_Response_Interface $response transaction response
+				 * @param Payment_Gateway_Direct $this instance
+				 */
+				$message = apply_filters( 'wc_payment_gateway_' . $this->get_id() . '_credit_card_transaction_approved_order_note', $message, $order, $response, $this );
 
-		}
+				$this->update_order_meta( $order, 'is_tender_type_card', true );
 
-		return $response;
+				$order->add_order_note( $message );
 
+			}
+
+			Performance_Logger::end( 'payment_transaction', $this->get_plugin() );
+			return $response;
+		} catch ( \Exception $e ) {
+			Performance_Logger::end( 'payment_transaction', $this->get_plugin(), true );
+			throw $e;
```

Comment:
And here

Reply:
This changes made only to log the performance do i don't think we needs to handle the exception here. WDYT?

Reply:
So looking at the method again, it already has a `@throws` statement so exception handling may already be taken care of downstream, which would mean nothing is needed here

Reply:
Yup. Thanks.